karma.conf.js

```diff
@@ -0,0 +1,75 @@
+module.exports = function (config) {
+  if (!process.env.SAUCE_USERNAME || !process.env.SAUCE_ACCESS_KEY) {
```

Comment:
Extract the SauceLabs specific parts that only get added if `process.env.SAUCE_USERNAME` and `SAUCE_ACCESS_KEY` is set would be a simple way to achieve it.